$OpenBSD: patch-src_libpiano_piano_c,v 1.8 2012/06/08 11:01:20 dcoppa Exp $

Fix ambiguous error message "Invalid Partner Login" on wrong
email address/password failure
(upstream git commit 81bf363ac69e0036562434ca0e7153c3e624c8c3)

More error messages
(upstream git commit 2d9b7bfd3395148419da27cd3c98de470a2d245b
 and commit b32348d88f700f4a3b5fb3a95f7ee40336952230)

--- src/libpiano/piano.c.orig	Sun May  6 16:33:54 2012
+++ src/libpiano/piano.c	Fri Jun  8 12:38:13 2012
@@ -250,6 +250,10 @@ const char *PianoErrorToStr (PianoReturn_t ret) {
 			return "Out of memory.";
 			break;
 
+		case PIANO_RET_INVALID_LOGIN:
+			return "Wrong email address or password.";
+			break;
+
 		/* pandora error messages */
 		case PIANO_RET_P_INTERNAL:
 			return "Internal error.";
@@ -281,6 +285,19 @@ const char *PianoErrorToStr (PianoReturn_t ret) {
 
 		case PIANO_RET_P_INVALID_PARTNER_LOGIN:
 			return "Invalid partner login.";
+			break;
+
+		case PIANO_RET_P_LICENSING_RESTRICTIONS:
+			return "Pandora is not available in your country. "
+					"Set up a control proxy (see manpage).";
+			break;
+
+		case PIANO_RET_P_PARTNER_NOT_AUTHORIZED:
+			return "Invalid partner credentials.";
+			break;
+
+		case PIANO_RET_P_LISTENER_NOT_AUTHORIZED:
+			return "Listener not authorized.";
 			break;
 
 		default:
